components/event-sources/reconciler/httpsource/controller.go

```diff
@@ -95,10 +121,61 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
 	})
 
+	authInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(sourcesv1alpha1.HTTPSourceGVK()),
+		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
+	})
+
 	// watch for changes to metrics/logging configs
 
 	cmw.Watch(metrics.ConfigMapName(), r.updateAdapterMetricsConfig)
 	cmw.Watch(logging.ConfigMapName(), r.updateAdapterLoggingConfig)
 
+	// wait for cache to sync
+	stop := make(chan struct{})
+	authFactory.Start(stop)
+	waitForInformersSyncOrDie(authFactory)
+
+	err := hasSynced(ctx, authFactory.WaitForCacheSync)
+	if err != nil {
+		log.Fatalf("Error waiting for caches sync: %s", err)
+	}
+
 	return impl
 }
+
+type waitForCacheSyncFunc func(stopCh <-chan struct{}) map[reflect.Type]bool
+
+// waitForInformersSyncOrDie blocks until all informer caches are synced, or panics after a timeout.
+func waitForInformersSyncOrDie(f authenticationinformersv1alpha1.SharedInformerFactory) {
+	ctx, cancel := context.WithTimeout(context.Background(), informerSyncTimeout)
+	defer cancel()
+
+	err := hasSynced(ctx, f.WaitForCacheSync)
+	if err != nil {
+		log.Fatalf("waiting for caches sync failed: %v", err)
+	}
+}
+
+// hasSynced blocks until the given informer sync waiting function completes. It returns an error if the passed context
+// gets canceled.
+func hasSynced(ctx context.Context, fn waitForCacheSyncFunc) error {
+	// synced gets closed as soon as fn returns
+	synced := make(chan struct{})
+
+	// closing stopWait forces fn to return, which happens whenever ctx
+	// gets canceled
+	stopWait := make(chan struct{})
+	defer close(stopWait)
+	go func() {
+		fn(stopWait)
+		close(synced)
+	}()
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case <-synced:
+	}
+	return nil
+}
```

Comment:
You can get rid of that entire code by delegating all the start/wait logic to `injection.Default.RegisterInformer()` ([godocs](https://pkg.go.dev/knative.dev/pkg/injection?tab=doc#InformerInjector)).

Reply:
Removed.